src/composite_core.jl

```diff
@@ -0,0 +1,124 @@
+struct PrimalAdditionFailedException{P} <: Exception
+    primal::P
+    differential::Composite{P}
+    original::Exception
+end
+
+function Base.showerror(io::IO, err::PrimalAdditionFailedException{P}) where {P}
+    println(io, "Could not construct $P after addition")
+    print("This probably means no default constructor (")
+    print(P, "(", join(propertynames(err.differential), ", "), ")")
+    println(") is defined.\n Either define a default constructor, or")
+    print("overload `ChainRulesCore.construct(::Type{$P}, ")
+    println(" ::$(typeof(err.differential)))`,")
+    println("or overload `Base.:+(::$P, ::$(typeof(err.differential)))`.")
+    println("Original Exception:")
+    print(io, err.original)
+    println()
+end
+
+
+
+"""
+    backing(x)
+
+Returns a version of `x` that is of the type the backing of a `Composite` is.
+i.e. as a `NamedTuple` or `Tuple`.
+
+This is kind of the opposite of `construct` for structs.
```

Comment:
Would be helpful if some context were added to this docstring. While it's reasonably clear what the functionality is, it's not really apparent from the docstring + code where you might use it. Or perhaps that belongs in the general documentation...

Reply:
yeah that is a good idea. It is for making everything into the same type as `Composite`'s `backing` field to simplify defining functions